src/braket/tasks/gate_model_quantum_task_result.py

```diff
@@ -278,13 +279,25 @@ def _from_object_internal_computational_basis_sampling(cls, result: GateModelTas
                 f"Measured qubits {measured_qubits} is not equivalent to number of qubits "
                 + f"{measurements.shape[1]} in measurements"
             )
-        result_types = (
-            result.resultTypes
-            if result.resultTypes
-            else GateModelQuantumTaskResult._calculate_result_types(
+        if result.resultTypes:
+            # Jaqcd does not return anything in the resultTypes schema field since the
+            # result types are easily parsable from the IR. However, an OpenQASM program
+            # specifies result types inline and parsing result types is more involved
+            # (ie. may involve dereferencing logical qubits at runtime), so the parsed
+            # result type specifications need to be returned, even if not calculated
+            # during simulation.
+            if not isinstance(result.resultTypes[0], ResultTypeValue):
```

Comment:
Not sure I understand what the new logic branch looks like; the old code already did the trick for OpenQASM programs sent to the service no? Is it possible to unify the results format?

Reply:
Good analysis here. The service got around this limitation by calculating result types on the service side for openqasm programs, in contrast to the jaqcd decision to calculate shots result types client side. We could take the same approach here, but then we would be duplicating the logic in this file in the simulator repo.

I thought it was a cleaner solution and more in line with the design decisions made for jaqcd circuits to simply pass back information about the result type types and add a branch here. It is a slight abuse of schema typing and may be confusing since we have several different IRs for tracking result types throughout the service. Nothing we can't fix with a union in our schemas lol

Reply:
That makes sense; also brings back to the table whether we should also allow device-defined behavior for shots>0.